modules/application-health-about.adoc

```diff
@@ -5,125 +5,165 @@
-kills and restarts the Container on the node.
+* HTTP GET. When using an HTTP GET probe, the probe determines the healthiness of the container by using a web hook. The probe is successful if the HTTP response code is between 200 and 399.
++
+You can use an HTTP GET probe with applications that return HTTP status codes when completely initialized.
 
-.Sample Liveness Check
+* Container Command. When using a Container Command probe, the probe executes a command inside the container. The probe is successful if the test exits with
+a `0` status.
+
+* TCP Socket. When using a TCP Socket probe, the probe attempts to open a socket to the container. The container is only
+considered healthy if the probe can establish a connection. You can use a TCP Socket probe with applications that do not start listening until
+initialization is complete.
+
+Probes have a number of fields that you can use to more precisely control the behavior of a probe:
+
+* `initialDelaySeconds`: The time, in seconds, after the container starts before the probe can be scheduled. The defaults is `0`.
+* `periodSeconds`: The delay, in seconds, between performing the probes. The default is `10`.
+* `timeoutSeconds`: The number of seconds of inactivity after which the probe times out and the container is assumed to have failed. The defaults is `1`.
+* `successThreshold`: The number of times that the probe must report success after a failure in order to reset the probe process. The value must be `1` for a Liveness probe. The default is `1`. 
+* `failureThreshold`: The number of times that the probe is allowed to fail. The defaults is 3. After the specified attempts: 
+** for a Liveness probe, the container is restarted 
+** for a Readiness probe, the pod is marked `Unready`
+** for a Startup probe, the container is killed and is subject to the pod's `restartPolicy`. 
++
+[NOTE]
+====
+The `timeoutSeconds` parameter has no effect on the Readiness and Liveness
+probes for Container Command probes. You can implement a timeout
+inside the probe itself, as {product-title} cannot time out on an exec call into
+the container. One way to implement a timeout in a probe is by using the `exec-timeout` command to run your
+Liveness or Readiness probes, as shown in the examples.
+====
+
+[discrete]
+[id="application-health-examples"]
+== Example probes
+
+The following are samples of different probes as they would appear in an object specification.  
+
+The following is an example of an HTTP GET Readiness probe:
+
+.Sample Readiness probe
 [source,yaml]
 ----
-apiVersion: v1
-kind: Pod
-metadata:
-  labels:
-    test: liveness
-  name: liveness-http
+...
 spec:
   containers:
-  - name: liveness-http
-    image: k8s.gcr.io/liveness <1>
+  - name: readiness <1>
     args:
-    - /server
-    livenessProbe: <2>
-      httpGet:   <3>
-        # host: my-host
-        # scheme: HTTPS
+    image: k8s.gcr.io/goproxy:0.1 <2>
```

Comment:
@weinliu @vlours What is the image that should be used? I grabbed these examples from the upstream. Is this image required?

Reply:
@lyman9966 ^^

Reply:
Hey @mburke5678, the image can be anything which can be use to run a container.

In fact, looking at the description "_**<2> The image to use for the Readiness probe.**_" (line **91**), I don't think this is appropriate, and may be confusing as the field image is not directly related to either Liveness or Readiness, it more the container image where we will apply the Liveness/Readiness rules. 
_Same on lines **125**, **160**, and line **49 in modules/application-health-configuring.adoc**_

From my point of view this should be something like:
"_**<2> The container image where to run the probe on.**_"
This will remove the confusion for not native english speaker which may assume that you have to set a special "liveness"/"readiness" image to run probes.

I hope my comment make sence 😉

Reply:
@vlours Thank you. That makes sense. Do I need to create the health check pod in the same project as the container I want to monitor?

Reply:
The Health check probes (liveness or readiness) are some functionalities of the POD, they will monitor the container status based on the check's type (exec, ..., ) you want to run. 
So they are part of the pod->container configuration. There is no notion of projects or dedicated health pods to run.
I hope my sentences have sense, I make it clear. 😅

Reply:
Ah! Looking at the Kubernetes docs, it looks like you create a separate pod spec for a health check [1]. That is not correct? The user adds a health check stanza to the pod spec for the container he wants to probe?  

[1] https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request

Reply:
Yes, this documentation is confusing. They should not use "liveness" for the pods, containers names 🙁
The description of [Define a TCP liveness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-tcp-liveness-probe) is a clearer example of a pod with a liveness probe.

Reply:
@vlours Thank you. 

Is this statement correct? it seems to be the behavior I am seeing:
_If you want to add or edit health checks in existing pod, you must edit the pod deployment configuration or use the *Developer* perspective in the web console._

Reply:
Hey Michael,

Yes, that looks correct. 
Where do you want to put this statement?